WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetailsViewModel.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import Yosemite
+
+/// View model for `CouponDetails` view
+///
+final class CouponDetailsViewModel: ObservableObject {
+    /// Code of the coupon
+    ///
+    @Published private(set) var couponCode: String = ""
+
+    /// Description of the coupon
+    ///
+    @Published private(set) var description: String = ""
+
+    /// Amount of the coupon
+    ///
+    @Published private(set) var amount: String = ""
+
+    /// Product limit for the coupon to be applied to
+    ///
+    @Published private(set) var applyTo: String = ""
+
+    /// Expiry date of the coupon
+    ///
+    @Published private(set) var expiryDate: String = ""
+
+    /// The current coupon
+    ///
+    private let coupon: Coupon
+
+    private let stores: StoresManager
+
+    init(coupon: Coupon,
+         stores: StoresManager = ServiceLocator.stores) {
+        self.coupon = coupon
+        self.stores = stores
+        populateDetails()
+    }
+}
+
+// MARK: - Private helpers
+//
+private extension CouponDetailsViewModel {
+    func populateDetails() {
+        couponCode = coupon.code
+        description = coupon.description
+
+        switch coupon.discountType {
+        case .percent:
+            let percentFormatter = NumberFormatter()
+            percentFormatter.numberStyle = .percent
+            if let amountDouble = Double(coupon.amount) {
+                let amountNumber = NSNumber(value: amountDouble / 100)
+                amount = percentFormatter.string(from: amountNumber) ?? ""
+            }
+        case .fixedCart, .fixedProduct:
+            let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
```

Comment:
nit: this isn't an issue in CI, but the currency in`ServiceLocator.currencySettings` might be non-USD from previous simulator runs for developers to run tests locally. It might be worth DI'ing `CurrencyFormatter` or `CurrencySettings` with a default using `ServiceLocator` and passing a value in tests so that the view model tests pass consistently

Reply:
Thanks for reminding me about that! Updated in 75d6618983194ab6b2e102b7e85cb7e2d40af403.